sdk/keyvault/keyvault/src/secretsModels.ts

```diff
@@ -64,15 +66,45 @@ export interface DeletedSecret extends Secret {
   readonly deletedDate?: Date;
 }
 
+/**
+ * NOTE: EntityVersion is an internal type used by Azure. To get versions, use
+ * the corresponding 'get' call for the entity (for example: `getSecret`). 
+ */
+export class EntityVersion {
+  private version: string;

```

Comment:
I'm not 100% sure about making an exported type for version.  This will give the user a better idea of what is expected in the case of APIs which take the version, but it does add another type to the package, and would have to be a type either duplicated or shared across the secrets, keys and certificate packages.  It's possible that adding a code snippet to the APIs which take in a version showing this pattern would be just as effective.

Reply:
Moving back to string for now. We can revisit later, but seems better to align with other languages and make the decision more unilaterally.